ql/src/java/org/apache/hadoop/hive/ql/parse/relnodegen/LateralViewPlan.java

```diff
@@ -114,7 +115,7 @@ public LateralViewPlan(ASTNode lateralView, RelOptCluster cluster, RelNode input
 
     this.lateralViewRel = HiveTableFunctionScan.create(cluster,
         TraitsUtil.getDefaultTraitSet(cluster), ImmutableList.of(inputRel), udtfCall,
-        null, retType, null);
+        null, retType, createColumnMappings(inputRel));
```

Comment:
I'm not sure I understand the question here, but lemme try to explain how I made this work.  I think it fits in with what is meant by column mappings...

The idea of the TableFunctionScan is to create multiple rows from a single row coming in.  The columns in the new rows can be broken down into 2 categories:

1) Columns with "exploded" or "created" data.  For instance, if the input row had a column of type array, each new row might contain an element from that array.
2) Columns directly from the inputRel.  This will be denormalized data and "joined" to the data created in the columns mentioned in 1)

The column mapping here depends on the columns mentioned in 2).  Thus, they will always map only to inputRel columns (from 2) and never to udtfCall columns (from 1).

Reply:
I was thinking that there is also a third category:
3. Columns from the `inputRel` that are transformed based on the UDTF and appear in the result (something that would enable the derived attribute in `RelColumnMapping`).

Anyways I don't fully understand how `RelColumnMapping` is supposed to work so I suppose we can leave with this mapping for the moment.